types/tables.d.ts

```diff
@@ -53,10 +53,17 @@ export interface ArcTable<Item = unknown> {
   update(params: UpdateParams<Item>, callback: Callback<UpdateOutput>): void;
 }
 
-export type ArcDB<Tables> = {
+type ArcDBWith<Tables> = {
   [tableName in keyof Tables]: ArcTable<Tables[tableName]>;
 };
 
+export type ArcDB<Tables> = ArcDBWith<Tables> & {
+  name(name: string): string;
+  reflect(): object;
```

Comment:
I think this could be more specific too. It could be:

```
 reflect(): {
    [tableName in keyof Tables]: string;
  };
  ```

Then the permissive assertion would be:

```
// $ExpectType { [x: string]: string; }
```

and the strict assertions would be:

```
  // $ExpectError
  db.reflect().wrong

  // $ExpectType string
  db.reflect().note
```

Reply:
Same here, I tested this locally and it works, and then you get auto-complete on the properties of the object returned by reflect.